### ISSUE TITLE
Nightly 2022-11-06 missing `std`?

### ISSUE THREAD

**Issue by Miridyan** [oxalica/rust-overlay#110]
I created a basic devshell with the example provided by the README, and I'm running into some strange issues. `flake.nix` is below:
```nix
{
  description = "A devShell example";

  inputs = {
    nixpkgs.url    = "github:NixOS/nixpkgs/nixos-unstable";
    rust-overlay.url = "github:oxalica/rust-overlay";
    flake-utils.url  = "github:numtide/flake-utils";
  };

  outputs = { self, nixpkgs, rust-overlay, flake-utils, ... }:
    flake-utils.lib.eachDefaultSystem (system:
      let
        overlays = [ (import rust-overlay) ];
        pkgs = import nixpkgs {
          inherit system overlays;
        };
      in
      with pkgs;
      {
        devShells.default = mkShell {
          buildInputs = [
            pkg-config
            (rust-bin.fromRustupToolchain { channel = "nightly-2022-11-05"; })
          ];
        };
      }
    );
}
```
Creating a devshell with this `flake.nix` file works just fine and the project compiles. However, if I change the channel to `nightly-2022-11-06` and run `nix develop` again, the project fails to compile due to missing `std`. I tried a few other channels including beta and stable and none of them had the missing std issue, only nightly-2022-11-06 exhibits this problem. 

I'm not sure if this is an issue with the overlay or the latest nightly, but it seemed strange that everything but that specific nightly seemed to be working fine.

**Comment by figsoda** [comment]
rust changed its sysroot resolution, fenix ran into the same problem and was just fixed: https://github.com/nix-community/fenix/pull/91 in case you want to try to fix it for rust-overlay

**Comment by oxalica** [comment]
Fixed since a1ce4b7c66ed9eb6fbefa88846e04a5341553ff2
